Implement libraqm for vector outputs

QuLogic opened the pull request:
## PR summary

This PR is a followup to #30000 which did raster backends, now for vector backends. I'm not 100% on the kerning calculation, but the resulting differences are quite small; I think this may because of using a 16.16 value vs a 26.6 creating minor rounding differences.

~~This is based on #29695 to reduce conflicts.~~

## PR checklist
- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

anntzer commented on src/ft2font_wrapper.cpp:
```diff
@@ -1409,6 +1409,119 @@ PyFT2Font__get_type1_encoding_vector(PyFT2Font *self)
+    const auto load_flags = static_cast<FT_Int32>(flags);
+
+    FT2Font::LanguageType languages;
+    if (auto value = std::get_if<FT2Font::LanguageType>(&languages_or_str)) {
```
Actually not really related to this PR, but is there a reason not to use the `std::visit(variant, overloaded {...})` pattern? (see example 4 of https://en.cppreference.com/w/cpp/utility/variant/visit2.html, or grep for `overloaded` in the mplcairo codebase)

QuLogic replied:
I've not tried `std::visit`, but at least for `std::get_if`, one reason why we stuck with it instead of something else is that it wouldn't compile on macOS with the current deployment target we have. It's possible that it's is fine now, but I don't believe we've bumped the target in a while.

QuLogic commented on src/ft2font_wrapper.cpp:
```diff
@@ -1409,6 +1409,119 @@ PyFT2Font__get_type1_encoding_vector(PyFT2Font *self)
+
+    .. warning::
+        This API uses the fallback list and is both private and provisional: do not use
+        it directly.
```
There would need to be some way to share the FreeType library itself, I think. One issue I had initially was between sharing the internal FreeType structures with another when I accidentally had `libraqm` link to an external one. I'm not sure how easy this would be, unless we moved to some common provider.

QuLogic commented:
Pushed two corrections: 1) SVG had lost the font features/languages setting and 2) fixed vertical offsets in PDF. Now we can do ridiculous accent stacks like this:
<img width="477" height="1266" alt="image" src="https://github.com/user-attachments/assets/4b07fdd0-c9b5-4c3e-89d1-74e041aac94e" />

QuLogic commented:
I've expanded the new feature tests to include `.eps` extension, since that's not there by default, and fixed issues in that format.

QuLogic commented:
I've rebased this and added the image changes for review now that there are no other PR dependencies. Most of the changes are pretty much invisible, as the glyph locations shift by <1/64 of a pixel due to https://github.com/matplotlib/matplotlib/pull/30000#issuecomment-2906695858 Sometimes this causes the rasterization for test comparison to shift a glyph by a whole pixel, since the DPI is rather low, but it's not too noticeable at a normal size. I don't know if we want to round to nearest 1/64th, or just run with this as is.

The main ones you should look at are `test_text/complex.*`, `test_text/features.*`, and `test_text/language.*`, which should all be consistent now with the PNG.

There are fortunately not too many image changes, but please don't merge with them included, as I'll move them to the other branch before doing so like the other PRs.

anntzer commented:
Sorry for not reviewing further, and congrats on getting this in.
